src/Foreign/Marshal/Alloc/Compat.hs

```diff
@@ -0,0 +1,44 @@
+#if !MIN_VERSION_base(4,8,0)
+{-# LANGUAGE ForeignFunctionInterface #-}
+#endif
+module Foreign.Marshal.Alloc.Compat (
```

Comment:
And this propagates up one more level to `Foreign`, which re-exports `Foreign.Marshal`.